client/my-sites/plans/controller.jsx

```diff
@@ -3,15 +3,37 @@
  */
 import page from 'page';
 import React from 'react';
-import { get } from 'lodash';
+import { get, omit } from 'lodash';
 
 /**
  * Internal Dependencies
  */
 import Plans from './plans';
 import { isValidFeatureKey } from 'lib/plans/features-list';
+import { getSelectedSite } from 'state/ui/selectors';
 
 export function plans( context, next ) {
+	// Users coming from signing up should see the checkout page. The signup flow redirects to the plans page
+	// assuming it will end up redirecting to the checkout page. This is a way of reducing the number of duplicate
+	// sites when users go back in the browser trying to change a plan during checkout.
+	// See https://github.com/Automattic/wp-calypso/issues/39424
+	const isComingFromSignUp = !! context.query.signup;
+	if ( isComingFromSignUp ) {
+		// Removes the signup query param so users going back to plans are not redirected to checkout again.
+		page.replace(
+			context.pathname,
+			{
+				...context.state,
+				query: omit( context.query, 'signup' ),
+			},
+			false,
+			false
+		);
+
+		const selectedSite = getSelectedSite( context.store.getState() );
```

Comment:
Could we use `getSelectedSiteSlug` instead, since we use only the slug from the site object?